drivers/soundwire/generic_bandwidth_allocation.c

```diff
@@ -48,6 +50,9 @@ void sdw_compute_slave_ports(struct sdw_master_runtime *m_rt,
 		slave_total_ch = 0;
 
 		list_for_each_entry(p_rt, &s_rt->port_list, port_node) {
+			if (p_rt->lane != t_data->lane)
+				continue;
+
```

Comment:
not following this one, in which cases would the 'continue' happen?

Reply:
When some p_rt uses lane 0 and the others uses lane 1. For example, headphone playback, headset capture, and speaker playback use lane 0 and DMIC capture uses lane 1.  We will have different param groups for different lanes. and _sdw_compute_port_params() will call sdw_compute_master_ports() -> sdw_compute_slave_ports() for each param group. And we only want to fill port params with the group param with the same lane.

Reply:
I don't think it's possible that different streams share the same m_rt, and that means they would be treated separately.

The only case where I think a stream can be split across lanes is by channels, e.g. channel 0..3 on lane1 and channel 4..7 on lane2.

That makes me wonder if the t_data-> lane is really meaningful... I've lost how this is supposed to work now.

Reply:
m_rt isn't shared by a stream, but sdw_group_params is. Please see below code in _sdw_compute_port_params().
```
	/* Run loop for all groups to compute transport parameters */
	for (i = 0; i < count; i++) {
		port_bo = 1;

		list_for_each_entry(m_rt, &bus->m_rt_list, bus_node) {
			sdw_compute_master_ports(m_rt, &params[i], port_bo, hstop);

			port_bo += m_rt->ch_count * m_rt->stream->params.bps;
		}

		hstop = hstop - params[i].hwidth;
	}
```

Once two lanes are used, we will have 2 sdw_group_params. And `list_for_each_entry(m_rt, &bus->m_rt_list, bus_node)` will be called for each sdw_group_params. So, sdw_compute_master_ports() will be called twice for each m_rt, one for `&params[0]` , and the other for `&params[1]`. As p_rts in a m_rt could use different lanes, so, we can't just return like
```
        if (rate != params->rate)
                return;
```
And we need to pass `params->lane` to sdw_compute_slave_ports() as we only want to fill xport params when `p_rt->lane == params->lane`. Because the params like hstart, hstop is not for the p_rt if the params->lane doesn't match p_rt->lane.

Reply:
@bardliao you really need to explain better in the commit message what groups are and why lanes should not share the groups.